src/sc_pipe_management/terra_to_portal_posting.py

```diff
@@ -415,7 +432,100 @@ def read_json_file(json_file_paths: str) -> dict:
+        list[PostResult]: list(PostResult(col_header='workflow_config_document', accession=IGVF UUID, error=error if any))
+    """
+    workflow_config_post_results = []
+    # All about posting the workflow config document
+    try:
+        # NOTE: A bit of hack here to get submission ID (RNA data will always be available)
+        terra_job_ids = parse_workflow_uuids_from_gs_path(
+            gs_path=terra_data_record['rna_kb_h5ad'], output_format='ids')
+        curr_workflow_config = api_tools.get_workflow_input_config(
+            terra_namespace=terra_namespace, terra_workspace=terra_workspace, submission_id=terra_job_ids['submission_id'], workflow_id=terra_job_ids['workflow_id'])
+        # Remove credential file path
+        curr_workflow_config['igvf_credentials'] = 'Google cloud path to a txt file with credentials to access the IGVF data portal.'
+        # Save config to a local file
+        json_output_dir = os.path.join(
+            output_dir, 'workflow_configs', terra_data_record['analysis_set_acc'])
+        local_file_path = dump_json(
+            input_json=curr_workflow_config, analysis_set_acc=terra_data_record['analysis_set_acc'], output_dir=json_output_dir)
+        # Make a document object payload
+        doc_aliases = f"igvf-dacc-processing-pipeline:{'_'.join(terra_job_ids.values())}_pipeline_config"
+        doc_payload = dict(
+            aliases=[
+                f"igvf-dacc-processing-pipeline:{'_'.join(terra_job_ids.values())}_pipeline_config"],
+            lab=lab,
+            award=award,
+            document_type='computational protocol',  # TODO: need a proper enum
+            description='Terra workflow configuration for the single-cell pipeline run',
+            attachment={'path': local_file_path},
+            _profile='document'
+        )
+        # Post the document to the portal
+        curr_post_res = single_post_to_portal(igvf_data_payload=doc_payload,
+                                              igvf_utils_api=igvf_utils_api,
+                                              upload_file=False)
+        workflow_config_post_results.append(PostResult.Success(
+            col_header='workflow_config_document', accession=curr_post_res))
+        # Patch the analysis set with the document accession
+        try:
+            anaset_patching_payload = {
+                'documents': [doc_aliases],
+                igvf_utils_api.IGVFID_KEY: f"/analysis-sets/{terra_data_record['analysis_set_acc']}/",
+                '_profile': 'analysis_set'
+            }
+            single_patch_to_portal(patching_payload=anaset_patching_payload,
+                                   igvf_utils_api=igvf_utils_api)
+            workflow_config_post_results.append(PostResult.Success(
+                col_header='analysis_set_workflow_config_patch', accession=terra_data_record["analysis_set_acc"]))
+        # Log error of patching analysis set with workflow config document
+        except Exception as e:
```

Comment:
In general one should never (say never..) be catching `Exception` and instead figure out what specifically we expect to go wrong here and catch that.